scripts/unicode.py

```diff
@@ -391,9 +391,19 @@ def gen_composition_table(canon_comp, out):
 def gen_decomposition_tables(canon_decomp, compat_decomp, cjk_compat_variants_decomp, out):
     tables = [(canon_decomp, 'canonical'), (compat_decomp, 'compatibility'), (cjk_compat_variants_decomp, 'cjk_compat_variants')]
     for table, name in tables:
-        gen_mph_data(name + '_decomposed', table, "(u32, &'static [char])",
-            lambda k: "(0x{:x}, &[{}])".format(k,
-                ", ".join("'\\u{%s}'" % hexify(c) for c in table[k])))
+        offsets = {}
+        offset = 0
+        out.write("pub(crate) const %s_DECOMPOSED_CHARS: &[char] = &[\n" % name.upper())
```

Comment:
Wouldn't a fixed array avoid the relocations altogether? As in, a `%s_DECOMPOSED_CHARS: [char; _] =` or so?

Reply:
The tables are const. The slice for the table themselves are not store in .data.rel.ro at all because of that. So there are actually no relocations for those (also, that would only be one per table, which is not as dramatic as one per entry).